x-pack/plugins/fleet/server/services/epm/package_service.ts

```diff
@@ -86,8 +87,17 @@ export class PackageServiceImpl implements PackageService {
   ) {}
 
   public asScoped(request: KibanaRequest) {
-    const preflightCheck = () => {
-      if (!checkSuperuser(request)) {
+    const preflightCheck = async (requiredAuthz?: FleetAuthzRouteConfig['fleetAuthz']) => {
+      if (requiredAuthz) {
+        const requestedAuthz = await getAuthzFromRequest(request);
+
+        const noRequiredAuthz = doesNotHaveRequiredFleetAuthz(requestedAuthz, requiredAuthz);
+        if (noRequiredAuthz === undefined || noRequiredAuthz === true) {
```

Comment:
added the `undefined` check because the [unit test](https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/services/epm/package_service.test.ts#L212) was failing, would be better to figure out how to create a kibana request with fake privileges